src/crypto/slow-hash.c

```diff
@@ -64,6 +64,19 @@ static void local_abort(const char *msg)
 #endif
 }
 
+static size_t get_num_slow_hash_iterations(void)
```

Comment:
This change only speeds up tests, at the cost of adding more runtime code to one of the most critical code paths. `getenv` is likely a linear search against the environment and is done for every `cn_slow_hash` call.

Did you run callgrind or similar to get instruction count additions here? I don't think its a huge amount, but yikes.

Reply:
I did not, but this called once before 500k looks (though the loop bound changes from an immedite to a variable), and once it's called once, it keeps the value in a static. I don't doubt it will be lost in the noise.

Reply:
As stated, I'm against adding more complications to this code path simply for increasing testing performance, since this is fairly critical.

Anyway - the static should be `std::atomic<size_t>` to remove platform dependent behavior (tearing) on reads/writes. Otherwise, the thread safety _seems_ iffy but is correct AFAIK (multiple threads should always compute the same value). `std::atomic<std::size_t> i{0}` will initialize at compile-time just as this code is doing.

Reply:
I forgot this was a C file -

Looks like there is already platform dependent `static` in this file - I guess it hasn't bitten yet. Although its a confusing case to debug if it does trigger. The read/write will _probably_ be atomic since `size_t` is usually paired with the processor word size. C11 atomic libs exist, but the interface is a bit more clunky.

Reply:
The worst it can do is calc twice, since the same is written. Unless this is actual UB, allowing the compiler to draw a troll face on the screen ?